### ISSUE TITLE
[Bug]: panic runtime error: invalid memory address or nil pointer dereference

### ISSUE THREAD

**Issue by domingozhang** [matrixorigin/matrixone#4616]
### Is there an existing issue for the same bug?

- [X] I have checked the existing issues.

### Environment

```Markdown
- Version or commit-id (e.g. v0.1.0 or 8b23a93):
- Hardware parameters:
- OS type:
- Others:
```

### Actual Behavior

SHOW CHARSET;
ERROR 1105 (HY000): panic runtime error: invalid memory address or nil pointer dereference: goroutine 214 [running]:
runtime/debug.Stack()
	/usr/local/go/src/runtime/debug/stack.go:24 +0x65
github.com/matrixorigin/matrixone/pkg/common/moerr.NewPanicError({0x521d8e0?, 0x61e6320})
	/var/root/matrixone/pkg/common/moerr/error.go:235 +0x65
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).ExecRequest.func1()
	/var/root/matrixone/pkg/frontend/mysql_cmd_executor.go:2108 +0x45
panic({0x521d8e0, 0x61e6320})
	/usr/loca

**Comment by daviszhen** [comment]
How to reproduce the bug?

**Comment by daviszhen** [comment]
any sql? any log?

**Comment by domingozhang** [comment]
SHOW CHARSET;

**Comment by domingozhang** [comment]
show character set where charset = 'utf8mb4';
ERROR 1105 (HY000): panic runtime error: invalid memory address or nil pointer dereference: goroutine 103 [running]:
runtime/debug.Stack()
	/usr/local/go/src/runtime/debug/stack.go:24 +0x65
github.com/matrixorigin/matrixone/pkg/common/moerr.NewPanicError({0x5229240?, 0x61fa2c0})
	/var/root/matrixone/pkg/common/moerr/error.go:235 +0x65
github.com/matrixorigin/matrixone/pkg/frontend.(*MysqlCmdExecutor).ExecRequest.func1()
	/var/root/matrixone/pkg/frontend/mysql_cmd_executor.go:2153 +0x45
panic({0x5229240, 0x61fa2c0})
	/usr/loca

**Comment by aressu1985** [comment]
fixed:777b7ebbe34b8d1d3fe0733de2db588a4fc40c71

mysql> show character set where charset = 'utf8mb4';
Empty set (0.01 sec)

mysql>
